server/src/main/java/org/elasticsearch/action/OriginalIndices.java

```diff
@@ -57,13 +58,20 @@ public IndicesOptions indicesOptions() {
     }
 
     public static OriginalIndices readOriginalIndices(StreamInput in) throws IOException {
+        if (in.getVersion().onOrAfter(Version.V_8_0_0) && in.readBoolean() == false) {
```

Comment:
Should we move the logic here to SearchShardTarget instead?

Reply:
sorry it's not needed. I pushed a change to cleanup the usage of search shard target when opening a new reader:
https://github.com/elastic/elasticsearch/pull/54966/commits/0ec84abf64beeae6c5dbb35217306e447ca725d8